node/pkg/wormchain/watcher.go

```diff
@@ -0,0 +1,517 @@
+				}
+
+				msgs := EventsToMessagePublications(txHash, events.Array(), logger)
+				for _, msg := range msgs {
+					e.msgChan <- msg
+					wormchainMessagesConfirmed.Inc()
+				}
+			}
+		}
+	}()
+
+	go func() {
+		defer close(errC)
+
+		for {
+			_, message, err := c.ReadMessage()
+
+
+			if err != nil {
+				p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDWormchain, 1)
+				wormchainConnectionErrors.WithLabelValues("channel_read_error").Inc()
+				logger.Error("error reading channel", zap.Error(err))
+				errC <- err
+				return
+			}
+
+			// Received a message from the blockchain
+			json := string(message)
+			
+			txHashRaw := gjson.Get(json, "result.events.tx\\.hash.0")
+			if !txHashRaw.Exists() {
+				logger.Warn("wormchain message does not have tx hash", zap.String("payload", json))
+				continue
+			}
+			txHash := txHashRaw.String()
+
+			events := gjson.Get(json, "result.data.value.TxResult.result.events")
+			if !events.Exists() {
+				logger.Warn("wormchain message has no events", zap.String("payload", json))
+				continue
+			}
+
+			msgs := EventsToMessagePublications( txHash, events.Array(), logger)
+			for _, msg := range msgs {
+				e.msgChan <- msg
+				wormchainMessagesConfirmed.Inc()
+			}
+
+			client := &http.Client{
+				Timeout: time.Second * 15,
+			}
+
+			// Query and report guardian set status
+			requestURL := fmt.Sprintf("%s/certusone/wormholechain/wormhole/latest_guardian_set_index", e.urlLCD)
+			req, err := http.NewRequestWithContext(ctx, http.MethodGet, requestURL, nil)
+			if err != nil {
+				p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDWormchain, 1)
+				wormchainConnectionErrors.WithLabelValues("guardian_set_index_req_error").Inc()
+				logger.Error("query latest guardian set index request error", zap.Error(err))
+				errC <- err
+				return
+			}
+
+			msm := time.Now()
+			resp, err := client.Do(req)
+			if err != nil {
+				p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDWormchain, 1)
+				logger.Error("query guardian set response error", zap.Error(err))
+				errC <- err
+				return
+			}
+
+			//TODO change this field name once the active set gets refactored to latest set
+			body, err := ioutil.ReadAll(resp.Body)
+			queryLatency.WithLabelValues("LatestGuardianSetIndex").Observe(time.Since(msm).Seconds())
+			if err != nil {
+				p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDWormchain, 1)
+				logger.Error("query guardian set error", zap.Error(err))
+				errC <- err
+				resp.Body.Close()
+				return
+			}
+
+			json = string(body)
+			guardianSetIndex := gjson.Get(json, "LatestGuardianSetIndex.index")
+			//TODO do we also need to retrieve the addresses via a second webcall, or can this be omitted?
+			//addresses := gjson.Get(json, "result.addresses.#.bytes")
+
+			logger.Debug("current guardian set on Wormchain",
+				zap.Any("guardianSetIndex", guardianSetIndex))
+			
+
+			resp.Body.Close()
+
+			// We do not send guardian changes to the processor - ETH guardians are the source of truth.
```

Comment:
We recently ripped the guardian set stuff out of the Terra / Injective watcher because it was only used for logging anyway. You can probably do the same thing here.

Reply:
Done